### PR TITLE
Xcode 10 Beta 5 AppKit

### PR DESCRIPTION
**Pull request by chamons**

### CODE REVIEW

File: src/foundation.cs [hunk 1]
```diff
@@ -8532,11 +8532,19 @@ interface NSObject2 : NSObjectProtocol {
...
 		NSObject GetDefaultPlaceholder (NSObject marker, NSString binding);
+#endif
 
 		[Static]
 		[Export ("setDefaultPlaceholder:forMarker:withBinding:")]
+#if XAMCORE_4_0 && MONOMAC
+		void SetDefaultPlaceholder (NSObject placeholder, NSBindingSelectionMarker marker, NSString binding);
+#else
 		void SetDefaultPlaceholder (NSObject placeholder, NSObject marker, NSString binding);
+#endif
```
Comment (spouliot) [inline src/foundation.cs]: it's weird that `marker` became `NSBindingSelectionMarker` here while it was `binding` in the previous one !?! logically I'd say the first one should be

```
NSObject GetDefaultPlaceholder (NSBindingSelectionMarker marker, NSString binding);
```

File: src/foundation.cs [hunk 2]
```diff
@@ -8597,6 +8605,25 @@ interface NSObject2 : NSObjectProtocol {
 		void AwakeFromNib ();
 	}
 
+#if MONOMAC // Defined in AppKit NSKeyValueBinding.h
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface NSBindingSelectionMarker : NSCopying
+	{
```
Comment (VincentDondain) [inline src/foundation.cs]: Minor: styling, curly brace goes up.

File: tests/xtro-sharpie/macOS-AppKit.ignore [hunk 3]
```diff
@@ -1147,3 +1148,7 @@
 ## DoCommandBySelector conflicts with NSTextViewDelegate in generated code
 !missing-protocol-member! NSTextInput::doCommandBySelector: not found
 
+## 42781537 Classes fail to conformsToProtocol despite header declaration
```
Comment (VincentDondain) [inline tests/xtro-sharpie/macOS-AppKit.ignore]: That's a lot of failing classes (;
`//radar`

File: tests/xtro-sharpie/macOS-AppKit.todo [hunk 4]
```diff
@@ -4,12 +4,3 @@
 !missing-selector! +NSColor::findHighlightColor not bound
 !missing-selector! +NSColor::placeholderTextColor not bound
 !missing-selector! NSSliderCell::setImage: not bound
-## appended from unclassified file
```
Comment (spouliot) [inline tests/xtro-sharpie/macOS-AppKit.todo]: what about the other todo (they are from past beta) ?

Comment (chamons) [reply tests/xtro-sharpie/macOS-AppKit.todo]: Unsure, they were not in this b4->b5 and I didn't see in b4 either. I'll look at them outside of this PR.

File: src/foundation.cs (hunk 1 above)
Comment (chamons) [reply src/foundation.cs]: So here's the diff in question:

```-+ (void)setDefaultPlaceholder:(nullable id)placeholder forMarker:(nullable id)marker withBinding:(NSBindingName)binding;    // marker can be nil or one of NSMultipleValuesMarker, NSNoSelectionMarker, NSNotApplicableMarker
-+ (nullable id)defaultPlaceholderForMarker:(nullable id)marker withBinding:(NSBindingName)binding;    // marker can be nil or one of NSMultipleValuesMarker, NSNoSelectionMarker, NSNotApplicableMarker
++ (void)setDefaultPlaceholder:(nullable id)placeholder forMarker:(nullable NSBindingSelectionMarker *)marker withBinding:(NSBindingName)binding;    // marker can be nil or one of multipleValuesSelectionMarker, noSelectionMarker, notApplicableSelectionMarker
++ (nullable id)defaultPlaceholderForMarker:(nullable NSBindingSelectionMarker *)marker withBinding:(NSBindingName)binding;    // marker can be nil or one of multipleValuesSelectionMarker, noSelectionMarker, notApplicableSelectionMarker
```

Comment (chamons) [reply src/foundation.cs]: In other words, I made a mistake.

Comment (spouliot) [reply src/foundation.cs]: so
```
+ (nullable id)defaultPlaceholderForMarker:(nullable NSBindingSelectionMarker *)marker withBinding:(NSBindingName)binding;    // marker can be nil or one of multipleValuesSelectionMarker, noSelectionMarker, notApplicableSelectionMarker
```
is not
```
NSObject GetDefaultPlaceholder (NSObject marker, NSBindingSelectionMarker binding);
```
but the binding I guessed, right ?

Comment (chamons) [reply src/foundation.cs]: I believe it is fixed now.

**Pull request merged**